src/components/content.js

```diff
@@ -718,9 +724,27 @@ class Content extends React.Component {
   render = () => {
     const { props } = this
     const { className, readOnly, state, tabIndex, role } = props
-    const { document } = state
+    const { document, selection } = state
+
+    let selectionNeedsRedraw = true
+
+    // If document and selection are the same we have stored after `onSelect`
+    // event, native selection does not need to be rendered again.
+    if (this.onSelectSelection) {
+      selectionNeedsRedraw = (
+        this.onSelectSelection.anchorKey !== selection.anchorKey ||
+        this.onSelectSelection.anchorOffset !== selection.anchorOffset ||
+        this.onSelectSelection.focusKey !== selection.focusKey ||
+        this.onSelectSelection.focusOffset !== selection.focusOffset ||
```

Comment:
We can't just compare identity of `selection` and `onSelectSelection` because the former is got by merging the ladder into the previous state selection.
So they are different objects even when they represent the same selection.
But, of course, we can use `Immutable.is`.